### PR TITLE
SONARXML-163 Support :: operator in XPathCheck

### PR DESCRIPTION
**Pull request by Wohops**

### CODE REVIEW

File: sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/checks/XPathCheckTest.java [hunk 1]
```diff
@@ -93,6 +93,13 @@ void test_with_default_namespace() {
     SonarXmlCheckVerifier.verifyIssues("with_default_namespaces.xml", getCheck("//template"));
   }
 
+  @Test
+  void test_with_double_colon() {
+    XPathCheck check = getCheck("//dependency/artifactId[starts-with(text(),'mule-http-connector')]//following-sibling::version[not(starts-with(text(),'1.6'))]");
```
Comment (dorian-burihabwa-sonarsource) [inline sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/checks/XPathCheckTest.java:98]: I feel like we should add a test with a path that actually contains `namespace-uri` or a single `:` for completeness of the test.

**Review by dorian-burihabwa-sonarsource — CHANGES_REQUESTED** [review]
This PR looks like it is fixing the problem

File: sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/checks/XPathCheckTest.java (hunk 1 above)
Comment (Wohops) [reply sonar-xml-plugin/src/test/java/org/sonar/plugins/xml/checks/XPathCheckTest.java:98]: pretty good point.

**Comment by Wohops** [comment]
@dorian-burihabwa-sonarsource I slightly changed the implementation of the solution to:
1. cache the result so we don't have to recompute it for every file
2. avoid usage of a stream for something that simple

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
LGTM thanks for the changes

**Pull request merged**